sdk/appconfiguration/app-configuration/src/internal/helpers.ts

```diff
@@ -166,39 +158,38 @@ export function makeConfigurationSettingEmpty(
  */
 export function transformKeyValue(kvp: KeyValue): ConfigurationSetting {
   const setting: ConfigurationSetting & KeyValue = {
+    value: undefined,
     ...kvp,
     isReadOnly: !!kvp.locked
   };
 
   delete setting.locked;
 
-  switch (setting.contentType) {
-    case featureFlagContentType: {
-      return deserializeFeatureFlag(setting) ?? setting;
-    }
-    case secretReferenceContentType: {
-      return deserializeSecretReference(setting) ?? setting;
-    }
-    default:
-      return setting;
-  }
+  return setting;
 }
 
 /**
  * @internal
  */
 export function serializeAsConfigurationSettingParam(
-  setting: FeatureFlagParam | SecretReferenceParam | ConfigurationSettingParam
+  setting:
+    | ConfigurationSettingParam
+    | ConfigurationSettingParam<FeatureFlagValue>
+    | ConfigurationSettingParam<SecretReferenceValue>
 ): ConfigurationSettingParam {
   switch (setting.contentType) {
```

Comment:
hmm... here I almost don't want this to be a switch but use UDTs to avoid having to cast manually

Reply:
Done, let me know how that looks!